### ISSUE TITLE
LAYOUT-CARD 1.3.1: Can'nt use - all custom cards has bad performance

### ISSUE THREAD

**Issue by zibous** [thomasloven/lovelace-layout-card#99]
With the new version of the home assistant, the following malfunction occurs when I use Layout Card:

- Each time I reload, if I use the LAYOUT-CARD, the constructor and the settings of each custom card are called several times.
- **Without LAYOUT-CARD only once**

Depending on how well the custom cards have been programmed, the loading process is very slow and uses a lot of memory.

**Requirement**:
 + Fresh install Home Assistant 0.117.1
 + Only one simple custom card and layout-card 1.3.1

**System**

arch | aarch64
-- | --
dev | false
docker | false
hassio | false
installation_type | Home Assistant Core
os_name | Linux
os_version | 5.8.0-1006-raspi
python_version | 3.8.6
timezone | Europe/Vienna
version | 0.117.1
virtualenv | true

Am I doing something wrong with the definition?
Is there a workaround for this?

**Test w/o layout-card 1.3.1**
```
SIMPLECART   0.0.1 simplecard.js:5:9
*****Event - card - constructor jfwft246a 1604152735189 simplecard.js:17:15
*****Event - card - setConfig oh8nbwuvr 1604152735190 simplecard.js:42:22
LAYOUT-CARD 1.3.1 IS INSTALLED <empty string> layout-card.js:92:145
```

**Test with layout-card 1.3.1**
```
SIMPLECART   0.0.1 simplecard.js:5:9
LAYOUT-CARD 1.3.1 IS INSTALLED <empty string> layout-card.js:92:145
*****Event - card - constructor 8vpv3pdyc 1604152656773 simplecard.js:17:15
*****Event - card - setConfig z7sgi2ym2 1604152656774 simplecard.js:42:22
*****Event - card - constructor n0z001lc3 1604152656790 simplecard.js:17:15
*****Event - card - setConfig yyxg9zy8r 1604152656790 simplecard.js:42:22
*****Event - card - constructor dr90385w3 1604152656879 simplecard.js:17:15
*****Event - card - setConfig mdq5h5rwn 1604152656880 simplecard.js:42:22
*****Event - card - constructor 4kef94bus 1604152656894 simplecard.js:17:15
*****Event - card - setConfig 4mosdvm02 1604152656895 simplecard.js:42:22
```

YAML:
```yaml
title: 'Testcase '
views:
  - path: default_view
    title: Home
    badges:
      - entity: binary_sensor.updater
      - entity: person.admin
      - entity: sun.sun
    cards:
      - type: 'custom:layout-card'
        layout: horizontal
        column_num: 1
        column_width: 1200
        max_width: 1200
        cards:
          - type: entities
            entities:
              - sensor.orangenbaum_light
              - sensor.orangenbaum_temperature
              - sensor.orangenbaum_moisture
              - sensor.orangenbaum_conductivity
            title: Garten
          - type: entities
            entities:
              - sensor.hacs
            title: System
          - type: weather-forecast
            entity: weather.home
            show_forecast: false
          - type: 'custom:simple-card'
            entities:
              - sun.sun

```

**Custom Card: simpleCard**
```javascript
/**  ----------------------------------------------------------------- 
TEST CASE SIMPLE CARD
/** -------------------------------------------------------------------*/

console.info(
        "%c SIMPLECART  %c ".concat("0.0.1", " "),
        "color: white; background: #2980b9; font-weight: 700;",
        "color: white; background: #e74c3c; font-weight: 700;"
);

class SimpleCard extends HTMLElement {
        constructor() {
            super();
            this.attachShadow({
                        mode: "open",
             });
             console.log(
                        "*****Event - card - constructor",
                        Math.random().toString(36).substr(2, 9),
                        new Date().getTime()
             );
        }

        set hass(hass) {
          // nothing happens here
        }

        createCard(){
            const eId = Math.random().toString(36).substr(2, 9);
            this.id = "card-" + eId;

            const card = document.createElement("ha-card");
            const content = document.createElement("div");
            content.id = "content-" + eId;
            content.style.cssText = "height:" +this.height+'px';

            card.appendChild(content);
            this.root.appendChild(card);
       }

       setConfig(config) {
            console.log(
                       "*****Event - card - setConfig",
                       Math.random().toString(36).substr(2, 9),
                       new Date().getTime()
            );
            
            this.root = this.shadowRoot;

            if (this.root.lastChild) this.root.removeChild(this.root.lastChild);
            
            this._config = config;
            this.height = this._config.height || 240;
            this.createCard();
       }

       getCardSize() {
         return 1;
       }

        getCardSize() {
          return 1;
        }

}
customElements.define("simple-card", SimpleCard);
```

This is working w/o any errors
```yaml
views:
  - title: Test 4
    cards:
      - type: vertical-stack
        cards:
          - type: 'custom:simple-card'
            title: 'Testcase 4: Card 1'
            entities:
              - sun.sun
          - type: horizontal-stack
            cards:
              - type: 'custom:simple-card'
                title: 'Testcase 4: Card 2'
                entities:
                  - sun.sun
              - type: 'custom:simple-card'
                title: 'Testcase 4: Card 3'
                entities:
                  - sensor.orangenbaum_light

```

**Comment by zibous** [comment]
@thomasloven 

It looks like it's based of the "card" item:

The constructor or the config is always called for each "**CARD**" item. 
In the case of a nested layout with several custom: layout cards, these are called several times.

**No custom:layout-card** Results: **calls 1x**
 - *****Event - card - setConfig Testcase 1: Card 1 31.10.2020, 19:13:57 1x simplecard.js:49:11
```yaml
views:
  - title: Test 1
    cards:
      - type: 'custom:simple-card'
        entities:
          - sun.sun
```

**1x type: 'custom:layout-card'**: Results: **calls 2x**
  - *****Event - card - setConfig Testcase 2: Card 1 31.10.2020, 19:13:59 2x simplecard.js:49:11
```yaml
views:
  - title: Test 2
    cards:
      - type: 'custom:layout-card'
        layout: horizontal
        max_columns: 3
        cards:
          - type: 'custom:simple-card'
            title: 'Testcase 2: Card 1'
            entities:
              - sun.sun		
```


**2x type: 'custom:layout-card'** :  Results calls: **4x**
    -  *****Event - card - setConfig Testcase 3: Card 1 31.10.2020, 19:14:01 4x simplecard.js:49:11
```yaml
views:
  - title: Test 3
    cards:
      - type: 'custom:layout-card'
        layout: vertical
        max_columns: 1
        cards:
          - type: entity
            entity: sensor.orangenbaum_light
          - type: 'custom:layout-card'
            layout: horizontal
            max_columns: 3
            cards:
              - type: 'custom:simple-card'
                title: 'Testcase 3: Card 1'
                entities:
                  - sun.sun
```

**Comment by thomasloven** [comment]
This should be fixed with 2.0.0.
Please confirm.
